libvast/src/partition_synopsis.cpp

```diff
@@ -55,39 +93,45 @@ void partition_synopsis::add(const table_slice& slice,
           syn->add(std::move(view));
       }
     };
-    auto key = qualified_record_field{layout, leaf_it->index};
-    if (!caf::holds_alternative<string_type>(leaf_it->field.type)) {
+    // Make a field synopsis if it was configured.
+    if (auto key = qualified_record_field{layout, leaf_it->index};
+        auto fprate = get_field_fprate(fp_rates, key)) {
       // Locate the relevant synopsis.
       auto it = field_synopses_.find(key);
       if (it == field_synopses_.end()) {
         // Attempt to create a synopsis if we have never seen this key before.
-        it = field_synopses_
-               .emplace(std::move(key), make_synopsis(leaf_it->field.type))
-               .first;
+        auto opts = synopsis_opts;
+        opts["string-synopsis-fp-rate"] = *fprate;
+        opts["address-synopsis-fp-rate"] = *fprate;
+        it
+          = field_synopses_
+              .emplace(std::move(key), make_synopsis(leaf_it->field.type, opts))
+              .first;
       }
       // If there exists a synopsis for a field, add the entire column.
       if (auto& syn = it->second)
         add_column(syn);
-    } else { // type == string
-      // All strings share a partition-wide synopsis.
-      // NOTE: if this is made configurable or removed, the pruning step from
-      // the catalog lookup must be adjusted acordingly.
-      field_synopses_[key] = nullptr;
-      // We need to prune the type's metadata here by converting it to a
-      // concrete type and back, because the type synopses are looked up
-      // independent from names and attributes.
-      auto prune = [&]<concrete_type T>(const T& x) {
-        return type{x};
-      };
-      auto cleaned_type = caf::visit(prune, leaf_it->field.type);
-      auto tt = type_synopses_.find(cleaned_type);
-      if (tt == type_synopses_.end())
-        tt = type_synopses_
-               .emplace(cleaned_type, make_synopsis(leaf_it->field.type))
-               .first;
-      if (auto& syn = tt->second)
-        add_column(syn);
+    } else {
+      // We still rely on having `field -> nullptr` mappings for all fields
+      // without a dedicated synopsis during lookup and .
+      field_synopses_.emplace(std::move(key), nullptr);
     }
+    // We need to prune the type's metadata here by converting it to a
+    // concrete type and back, because the type synopses are looked up
+    // independent from names and attributes.
+    auto prune = [&]<concrete_type T>(const T& x) {
+      return type{x};
+    };
+    auto cleaned_type = caf::visit(prune, leaf_it->field.type);
```

Comment:
I don't see a fundamental reason that prevents us from supporting aliases in `get_type_fprate`, we would just have to call it here instead of before the loop. You decide if this should be part of this PR or another one :).

Reply:
Let's do this clean-slate with the next-gen partition FlatBuffers table. It needs to touch this anyways, so there's no need to risk this now.

Reply:
I don't think that extension would require a change to the table, but I'm fine with doing it at a later time too.

Reply:
I'm not sure I understand the comment, we already do the type synopsis lookup using the cleaned type so if that resolves aliases then everything should work automatically?